subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/IndexPageGenerator.java

```diff
@@ -16,96 +16,238 @@
+        }
+
+        return scenario;
+    }
 
-    public IndexPageGenerator(List<NavigationItem> navigationItems) {
-        this.navigationItems = navigationItems;
+    private ScenarioBuildResultData.ExecutionData extractExecutionData(PerformanceTestExecution performanceTestExecution) {
+        List<MeasuredOperationList> nonEmptyExecutions = performanceTestExecution
+            .getScenarios()
+            .stream()
+            .filter(testExecution -> !testExecution.getTotalTime().isEmpty())
+            .collect(Collectors.toList());
+        if (nonEmptyExecutions.size() > 1) {
+            return new ScenarioBuildResultData.ExecutionData(performanceTestExecution.getStartTime(), getCommit(performanceTestExecution), nonEmptyExecutions.get(0), nonEmptyExecutions.get(1));
+        } else {
+            return null;
+        }
+    }
+
+    private String getCommit(PerformanceTestExecution execution) {
+        if (execution.getVcsCommits().isEmpty()) {
+            return "";
+        } else {
+            String commit = execution.getVcsCommits().get(0);
+            return commit.substring(0, Math.min(7, commit.length()));
+        }
     }
 
     @Override
     public void render(final ResultsStore store, Writer writer) throws IOException {
-        new MetricsHtml(writer) {{
-            html();
-                head();
-                    headSection(this);
-                    title().text("Profile report for channel " + ResultsStoreHelper.determineChannel()).end();
-                end();
-                body();
-
-                navigation(navigationItems);
-
-                div().id("content");
-
-                    Map<String, String> archived = new LinkedHashMap<String, String>();
-                    List<String> testNames = store.getTestNames();
-                    div().id("controls").end();
-                    for (String testName : testNames) {
-                        PerformanceTestHistory testHistory = store.getTestResults(testName, 5, 14, ResultsStoreHelper.determineChannel());
-                        List<? extends PerformanceTestExecution> results = testHistory.getExecutions();
-                        results = filterForRequestedCommit(results);
-                        if (results.isEmpty()) {
-                            archived.put(urlEncode(testHistory.getId()), testHistory.getDisplayName());
-                            continue;
+        new MetricsHtml(writer) {
+            // @formatter:off
+            {
+                html();
+                    head();
+                        headSection(this);
+                        link().rel("stylesheet").type("text/css").href("https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css").end();
+                        script().src("https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/js/bootstrap.min.js").end();
```

Comment:
Why aren't we using the latest bootstrap version (`4.1.3`)?

Reply:
I wasn't aware of it... at that point the top articles of google are all `4.0.0` so I thought it's the latest. Changed to `4.1.3`.